### ISSUE TITLE
Compendium entries grouping/sorting by their Type

### ISSUE THREAD

**Issue by strusik01** [fmasa/wfrp-master#382]
**Is your feature request related to a problem? Please describe.**
The Compendium entries, like spells or trappings, are listed in alphabetical order only. It's hard for my players to browse specific types of entires.

**Describe the solution you'd like**
It would be great to have all entries grouped by their type, then alphabetically.
For example trappings list could be grouped as Weapons, Armour, Tools etc., spells as Fire, Light etc., and Careers as Couriers, Warriors etc.

If sorting by Types, it would also be great to include sorting by subtypes as well. Example: Trappings sorted by types AND Weapons sorted by weapons groups.

**Describe alternatives you've considered**
Even better (though harder to implement I imagine) would be a proper "Sort by" function. Sorting by Type, Name (Alphabeting A-Z / Z-A) would be very comfortable. 

Various Compendium categories could even have their specific sorting options, like for Spells - sorting by CN, or for Trappings - sorting by their price.

**Comment by fmasa** [comment]
@strusik01 would UI similar to Journal work here? Having Trappings grouped by Type (e.g. Melee Weapons) and subtype (e.g. Fencing), Spells grouped by Lore, Miracles by God?

![Image](https://github.com/user-attachments/assets/14bf788a-10d0-495a-8f55-12c8c7307f89)

In addition showing basic metadata within the list (CN for spells, price for Trappings, etc).

**Comment by strusik01** [comment]
@fmasa Yes, I believe it would work well, if their respective icons will still be displayed as they are now in Compendium
